Automattic/wp-calypso
# Earn: Convert main component to typescript

## Proposed Changes

Converts Earn > Main to TypeScript. 

## Testing Instructions

1) Checkout this branch, go to `http://calypso.localhost:3000/earn/YOURDOMAIN`, and click around different tabs and sections to confirm navigation still works the same as before. You'll want to connect Stripe if you haven't so that you can navigate between sections behind the Connect to Stripe screen. 

2) Confirm there are no TS errors in vscode.

## Pre-merge Checklist
